File: src/main/java/gargoyle/netsquares/store/prop/PrefDoubleProperty.java
package gargoyle.netsquares.store.prop;

public class PrefDoubleProperty extends BasePrefProperty<Double> {
    public PrefDoubleProperty(Class<?> pkg, String key, Double defaultValue) {
        super(pkg, key, defaultValue);
    }

    @Override
    protected void _set(Double value) {
        preferences.putDouble(key, value);
    }

    public Double get(Double defaultValue) {
        return preferences.getDouble(key, defaultValue);
    }

    public void setValue(double value) {
        set(value);
    }
}

File: src/main/kotlin/gargoyle/netsquares/store/prop/PrefDoubleProperty.kt
package gargoyle.netsquares.store.prop

import kotlin.reflect.KClass

class PrefDoubleProperty(pkg: KClass<*>, key: String, defaultValue: Double) :
    BasePrefProperty<Double>(pkg, key, defaultValue) {
    override fun _set(value: Double) {
        preferences.putDouble(key, value)
    }

    override fun get(defaultValue: Double): Double {
        return preferences.getDouble(key, defaultValue)
    }

    fun setValue(value: Double) {
        set(value)
    }
}
